app/Repositories/ServiceRepository/Variable.php

```diff
@@ -60,28 +60,29 @@ public function create($id, array $data)
             throw new DisplayException('The default value you entered cannot violate the regex requirements.');
         }
 
-        if (Models\ServiceVariables::where('env_variable', $data['env_variable'])->where('option_id', $option->id)->first()) {
+        if (Models\ServiceVariable::where('env_variable', $data['env_variable'])->where('option_id', $option->id)->first()) {
             throw new DisplayException('An environment variable with that name already exists for this option.');
         }
 
         $data['user_viewable'] = (isset($data['user_viewable']) && in_array((int) $data['user_viewable'], [0, 1])) ? $data['user_viewable'] : 0;
         $data['user_editable'] = (isset($data['user_editable']) && in_array((int) $data['user_editable'], [0, 1])) ? $data['user_editable'] : 0;
         $data['required'] = (isset($data['required']) && in_array((int) $data['required'], [0, 1])) ? $data['required'] : 0;
+        $data['option_id'] = $option->id;
 
-        $variable = new Models\ServiceVariables;
-        $variable->option_id = $option->id;
-        $variable->fill($data);
+        $variable = Models\ServiceVariable::create($data);
 
-        return $variable->save();
+        return $variable;
     }
 
     public function delete($id)
     {
-        $variable = Models\ServiceVariables::findOrFail($id);
+        $variable = Models\ServiceVariable::with('ServerVariable')->findOrFail($id);
```

Comment:
Why `ServerVariable` and not `serverVariable`? Arent the other ones lowercase? (not sure)